lib/thunkMiddleware.lua

```diff
@@ -0,0 +1,17 @@
+--[[
+	A middleware that allows for functions to be dispatched.
+	Functions will receive a single argument, the store itself.
+	This middleware consumes the function; middleware further down the chain
+	will not receive it.
+]]
+local function thunkMiddleware(next)
+	return function(store, action)
+		if typeof(action) == "function" then
+			action(store)
```

Comment:
We should make sure to return the value that `action` returns here, and write a test for it.

With Redux thunks, it's common to return a Promise, which gets passed back to whoever dispatched the action.

```lua
store:dispatch(someAsync())
    :andThen(function(result)
        print("My async action completed!")
    end)
```

We might have to tweak the middleware implementation (and potentially the loggerMiddleware) to support this case, as well!

Reply:
This shouldn't need a change to middlewares, but it will require a change to `loggerMiddleware`. The underlying middleware design should be okay as-is.

This does raise the question: should `dispatch` return a value? In Redux, it returns the dispatched action; right now it doesn't return anything.

Reply:
I don't really see much of a use to making `dispatch` return a value other than what some middleware would return. Do you know why Redux does that?

Reply:
The documentation explains it as "for convenience"; I'm not aware of any more detailed explanation. In the absence of that, this should be ready to merge.